model/waterfall.go

```diff
@@ -270,7 +264,21 @@ func GetActiveWaterfallVersions(ctx context.Context, projectId string, opts Wate
 	pipeline := []bson.M{}
 
 	if len(opts.Variants) > 0 {
-		buildVariantPipeline, err := getBuildVariantFilterPipeline(ctx, opts.Variants, match, projectId)
+		var bvSearchCutoff int
+		if pagingForward {
+			bvSearchCutoff = opts.MaxOrder - MaxWaterfallVersionLimit
+		} else if pagingBackward {
+			// When paginating backwards, the order specifies the oldest version that will be investigated in the query. No need to increment by MaxWaterfallVersionLimit
+			bvSearchCutoff = opts.MinOrder
+		} else {
+			mostRecentVersion, err := GetMostRecentWaterfallVersion(ctx, projectId)
```

Comment:
Is this on first load?

Reply:
It runs when the Waterfall query is called! Which is on first load and on polling updates